src/Build/BackEnd/Components/Communications/NodeProviderOutOfProcBase.cs

```diff
@@ -275,7 +275,10 @@ protected NodeContext GetNode(string msbuildLocation, string commandLineArgs, in
 
                 // Create the node process
                 Process msbuildProcess = LaunchNode(msbuildLocation, commandLineArgs);
-                _processesToIgnore.Add(GetProcessesToIgnoreKey(hostHandshake, msbuildProcess.Id));
+                if (_componentHost.BuildParameters.EnableNodeReuse)
+                {
+                    _processesToIgnore.Add(GetProcessesToIgnoreKey(hostHandshake, msbuildProcess.Id));
```

Comment:
> Please fix those few concurrency bugs I have mentioned in comments.
> 
> Why is this change isolated to `/nr:false` path only? Since `nodeReuse` is on by default, we are missing a big customer segment with this optimization.

I have tried those changes and hit threading issues, so I have scoped down the feature to /nr:false.  Our development environment has /nr:false in the build scripts so to us, it is the default.  @Forgind PR #4905 is more comprehensive to address nodeReuse.